.github/workflows/deploy-azure-function.yml

```diff
@@ -0,0 +1,68 @@
+# TODO: Add support for deploying multiple apps
+
+name: Deploy to Azure Function
+on:
+  workflow_call:
+    inputs:
+      functionapp-name:
+        description: "The name of the function app"
+        required: true
+        type: string
+      resource-group:
+        description: "The name of the Azure resource group"
+        required: false
+        type: string
+      bicep-file:
+        description: "Path to bicep rules file"
+        type: string
+        default: ./infrastructure/addSecurityRestriction.bicep
+      
+    secrets:
+      token:
+        required: false
```

Comment:
Deploy will fail without credentials so this should maybe be `required: true` ?

Reply:
The reason it is not required is this workflow is being reused for testing/prod deployment. For testing deployment we don't use it though as we don't log in to Azure - that what makes it obligatory only for prod deployment

Reply:
Aha, gotcha .. only used for the network rules. In which case, no comments.